onnx/model_container.py

```diff
@@ -0,0 +1,324 @@
+            if prop.value not in self.large_initializers:
+                raise RuntimeError(
+                    f"Unable to find large tensor named {tensor.name!r} "
+                    f"with location {prop.value!r} in "
+                    f"{sorted(self.large_initializers)}."
+                )
+
+    def _save_external(
+        self, file_path: str, all_tensors_to_one_file: bool
+    ) -> onnx.ModelProto:
+        """
+        Save the large model into a main onnx file and one file
+        per tensor. Follows the same format as :func:`write_external_data_tensors
+        <onnx.external_data_helper.write_external_data_tensors>`.
+        The main model needs to be modified to update the file location,
+        the function returns this modified copy.
+
+        Arguments:
+            file_path: model file
+            all_tensors_to_one_file: all tensors in one file
+
+        Returns:
+            modified main model proto
+        """
+        _unique_names: set[str] = set()
+
+        def _clean_name(prefix: str, name: str) -> str:
+            for c in ":/\\;,!":
+                name = name.replace(c, "")
+            base_name = name
+            i = 0
+            while name in _unique_names:
+                i += 1
+                name = f"{base_name}_{i}"
+            _unique_names.add(name)
+            return name
+
+        folder = os.path.dirname(file_path)
+        if not os.path.exists(folder):
+            raise FileNotFoundError(f"Folder {folder!r} does not exist.")
+        proto = self.model_proto.SerializeToString()
+        copy = onnx.ModelProto()
+        copy.ParseFromString(proto)
+        prefix = os.path.splitext(os.path.split(file_path)[-1])[0]
+
+        if all_tensors_to_one_file:
+            file_weight = f"{os.path.split(file_path)[1]}.weight"
+            full_file_weight = f"{file_path}.weight"
+            offset = 0
+            with open(full_file_weight, "wb") as f:
+                pass
+
+        for tensor in ext_data._get_all_tensors(copy):
+            if not ext_data.uses_external_data(tensor):
+                continue
+            prop: onnx.StringStringEntryProto | None = None
+            for ext in tensor.external_data:  # type: ignore[assignment]
+                if ext.key == "location":  # type: ignore[attr-defined]
+                    prop = ext  # type: ignore[assignment]
+            if prop is None:
+                raise RuntimeError(
+                    f"No location found for tensor name {tensor.name!r}."
+                )
+            if prop.value not in self.large_initializers:
+                raise RuntimeError(
+                    f"Unable to find large tensor named {tensor.name!r} "
+                    f"with location {prop.value!r} in "
+                    f"{sorted(self.large_initializers)}."
+                )
+            np_tensor = self.large_initializers[prop.value]
+
+            if all_tensors_to_one_file:
+                buffer = np_tensor.tobytes()
```

Comment:
I think when writing to file, we need to convert to little-endian ordering. This is my understanding of the proto specification (though it is a bit confusing; I am basing this on my interpretation of [this proto line](https://github.com/onnx/onnx/blob/6de4d5be64a0ad7d486061b6c92975a038518bf6/onnx/onnx.in.proto#L586C35-L586C54) and the existing code the reads/writes the file from raw_data directly.

Reply:
So, we should call `byteswap` on `np_tensor` if it is big-endian?